src/components/shapes/calc_autorange.js

```diff
@@ -23,22 +23,18 @@ module.exports = function calcAutorange(gd) {
 
         // paper and axis domain referenced shapes don't affect autorange
         if(shape.xref !== 'paper' && xRefType !== 'domain') {
-            var vx0 = shape.xsizemode === 'pixel' ? shape.xanchor : shape.x0;
-            var vx1 = shape.xsizemode === 'pixel' ? shape.xanchor : shape.x1;
             ax = Axes.getFromId(gd, shape.xref);
 
-            bounds = shapeBounds(ax, vx0, vx1, shape.path, constants.paramIsX);
+            bounds = shapeBounds(ax, shape, constants.paramIsX, false);
```

Comment:
After considering https://github.com/plotly/plotly.js/pull/7010/files#r1626065612
```suggestion
            bounds = shapeBounds(ax, shape, constants.paramIsX);
```